### PR TITLE
Adds SOC4 test and fixes in contconic

### PR DESCRIPTION
**Pull request by joaquimg**

### CODE REVIEW

**Comment by joaquimg** [comment]
see list in #23

File: test/contconic.jl [hunk 1]
```diff
@@ -503,7 +503,71 @@ function contconictest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
...
+
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), c1)
+            c1_dual = MOI.getattribute(m, MOI.ConstraintDual(), c1) 
+
+            @test dot(c,x_primal) ≈ -dot(c1_dual,b) atol=ε
+            @test norm((c+A'c1_dual) - x_dual) ≈ 0.0 atol=ε            
+        end
+    end
+
+    if MOI.supportsproblem(solver, MOI.ScalarAffineFunction{Float64}, [(MOI.ScalarAffineFunction{Float64}, MOI.EqualTo{Float64}), (MOI.VectorVariablewiseFunction, MOI.PositiveSemidefiniteConeTriangle), (MOI.VectorVariablewiseFunction, MOI.SecondOrderCone)])
```
Comment (blegat) [inline test/contconic.jl]: I have just noticed: here I use `MOI.EqualTo{Float64}` but in the doc and in other tests, we use `MOI.EqualTo`, `MOI.GreaterThan` and `MOI.LessThan`. Why do we drop the type there ?

Comment (joaquimg) [inline test/contconic.jl]: Good question, I think its strange to use parameters in ScalarAffineFuncition and not there. Any ideas @mlubin?

Comment (mlubin) [inline test/contconic.jl]: it should be `MOI.EqualTo{Float64}` everywhere

File: test/contconic.jl [hunk 2]
```diff
@@ -506,6 +506,70 @@ function contconictest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
...
+            @test MOI.cangetattribute(m, MOI.PrimalStatus())
+            @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+            @test MOI.cangetattribute(m, MOI.DualStatus())
+            @test MOI.getattribute(m, MOI.DualStatus()) == MOI.FeasiblePoint
+
+            @test MOI.cangetattribute(m, MOI.VariablePrimal(), x)
+            x_primal = MOI.getattribute(m, MOI.VariablePrimal(), x)
+            @test x_primal[1]^2 ≥ x_primal[4]^2 + x_primal[5]^2 - ε
+
+            @test MOI.cangetattribute(m, MOI.VariableDual(), x)
```
Comment (mlubin) [inline test/contconic.jl]: `VariableDual` is not a thing anymore. It should be `ConstraintDual` on the reference `c2`.

**Pull request merged**